code/modules/mob/living/simple_animal/shade.dm

```diff
@@ -52,3 +53,186 @@
+	icon = 'icons/obj/projectiles.dmi'
+	icon_state = "bluespace"
+	icon_living = "bluespace"
+	icon_dead = "bluespace"
+	maxHealth = 100
+	health = 100
+	universal_speak = 1
+	universal_understand = 1
+	speak_emote = list("echoes")
+	emote_hear = list("echoes","echoes")
+	response_help  = "brushes against"
+	response_disarm = "brushes against"
+	response_harm   = "brushes against"
+	attacktext = "brushes against"
+	melee_damage_lower = 0
+	melee_damage_upper = 0
+	status_flags = CANPUSH
+	heat_damage_per_tick = 0
+	cold_damage_per_tick = 0
+	unsuitable_atoms_damage = 0
+	incorporeal_move = 3
+	mob_size = 0
+	density = 0
+	speed = 1
+	var/last_message_heard
+	var/message_countdown = 300
+	var/heard_dying_message = 0
+
+/mob/living/simple_animal/shade/bluespace/apply_damage()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/adjustBruteLoss()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/adjustFireLoss()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/adjustToxLoss()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/adjustOxyLoss()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/adjustHalLoss()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/attempt_grab()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/ex_act()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/crush_act()
+	return 0
+
+/mob/living/simple_animal/shade/bluespace/Initialize()
+	. = ..()
+	set_light(2, 1, l_color = LIGHT_COLOR_CYAN)
+
+/mob/living/simple_animal/shade/bluespace/Stat()
+	..()
+
+	if(statpanel("Status"))
+		stat(null, "Strength of Echoes: [message_countdown]")
+
+/mob/living/simple_animal/shade/bluespace/updatehealth()
+	message_countdown = max(0, message_countdown - 1)
+	if(message_countdown <= 0)
+		adjustCloneLoss(1)
+		if(!heard_dying_message)
+			heard_dying_message = 1
+			to_chat(src, "<span class='danger'>You feel yourself begin to fade away!</span>")
+	..()
+
+/mob/living/simple_animal/shade/bluespace/hear_say(var/message, var/verb = "says", var/datum/language/language = null, var/alt_name = "", var/italics = 0, var/mob/speaker = null, var/sound/speech_sound, var/sound_vol)
+	..()
+	if(speaker && speaker != src)
+		last_message_heard = message
+		message_countdown = min(300, message_countdown + 50)
+		health = min(maxHealth, health + 5)
+		if(heard_dying_message)
+			heard_dying_message = 0
+			to_chat(src, "<span class='notice'>The soothing echoes of life reinvigorate you.</span>")
+
+/mob/living/simple_animal/shade/bluespace/say(var/message)
+	var/list/words_in_memory = dd_text2List(last_message_heard, " ")
+	var/list/words_in_message = dd_text2List(message, " ")
+	for(var/word1 in words_in_message)
+		var/valid = 0
+		for(var/word2 in words_in_memory)
+			if(lowertext(word1) != lowertext(word2))
+				continue
+			else
+				valid = 1
+				break
+		if(!valid)
+			message = replacetext(message, word1, pick(words_in_memory))
+	message = slur(message,rand(10, 20))

```

Comment:
This might be a little redundant. Slur already is randomized. slur(message,15) would work just as well.